src/modules/ipc.cpp

```diff
@@ -59,11 +60,28 @@ namespace modules {
     for (auto& hook : m_hooks) {
       hook->command = pid_token(hook->command);
     }
-    m_formatter->add(DEFAULT_FORMAT, TAG_OUTPUT, {TAG_OUTPUT});
 
-    for (size_t i = 0; i < m_hooks.size(); i++) {
-      string format_i = "format-" + to_string(i);
-      m_formatter->add_optional(format_i, {TAG_OUTPUT});
+    m_formatter->add(DEFAULT_FORMAT, TAG_LABEL, {TAG_LABEL});
+
+    m_format = DEFAULT_FORMAT;
+    if (m_formatter->has(TAG_OUTPUT, DEFAULT_FORMAT)) {
```

Comment:
You don't need to load separate formats based on whether the user uses `<output>`. You can simply add `<output>` to the whitelist when adding the default format (`{TAG_LABEL, TAG_OUTPUT}`).

That way, you don't need to duplicate the code in the if and else branches